File: src/slash-commands/msgverif.js
import { SlashCommandBuilder, ActionRowBuilder, ButtonBuilder, ButtonStyle, PermissionsBitField } from 'discord.js';
/*
 * Peluche Bot — programme personnel de Electro / MathéoCASSY
 * https://github.com/MatheoCASSY/
 */
export default {
  data: new SlashCommandBuilder()
    .setName('msgverif')
    .setDescription('Publie un message avec un bouton pour renvoyer le message de vérification'),
  async execute(interaction) {
  // Permission: ManageGuild or VERIFIER_ROLE env
  const verifierRoleRaw = process.env.VERIFIER_ROLE;
  const verifierRole = (typeof verifierRoleRaw === 'string' ? verifierRoleRaw.trim() : verifierRoleRaw);
    const member = interaction.member;
    let allowed = false;
    if (member.permissions && member.permissions.has(PermissionsBitField.Flags.ManageGuild)) allowed = true;
    if (!allowed && verifierRole) {
      const r = interaction.guild.roles.cache.get(verifierRole) || interaction.guild.roles.cache.find(x => x.name === verifierRole);
      if (r && member.roles.cache.has(r.id)) allowed = true;
    }
    if (!allowed) {
      await interaction.reply({ content: "Vous n'êtes pas autorisé à utiliser cette commande.", ephemeral: true });
      return;
    }

    const button = new ButtonBuilder()
      .setCustomId('request_verif')
      .setLabel('Recevoir le message de vérification')
      .setStyle(ButtonStyle.Primary);

    const row = new ActionRowBuilder().addComponents(button);

    // Send the visible button message to the channel but avoid leaving a reply to the user
    try {
      // Acknowledge the interaction silently (ephemeral) then post the actual message to the channel
      await interaction.deferReply({ ephemeral: true }).catch(() => {});
      if (interaction.channel) {
        await interaction.channel.send({ content: 'Cliquez sur le bouton ci-dessous pour recevoir le message de vérification en DM.', components: [row] });
      } else {
        await interaction.followUp({ content: 'Impossible d\'envoyer le message de vérification ici (canal introuvable).', ephemeral: true }).catch(() => {});
      }
    } catch (err) {
      await interaction.followUp({ content: 'Erreur lors de l\'envoi du message de vérification.', ephemeral: true }).catch(() => {});
    }
    // Remove the ephemeral acknowledgement so the user doesn't see any reply
    try { await interaction.deleteReply().catch(() => {}); } catch (e) { /* ignore */ }
  }
};

File: Python/slash_commands/msgverif.py
"""/msgverif - publish a button message to allow users to request verification DM."""
import os
from logger import Logger
logger = Logger()
name = 'msgverif'
description = 'Publie un message avec un bouton pour renvoyer le message de vérification'

async def execute(interaction, **kwargs):
    try:
        verifier_role = os.getenv('VERIFIER_ROLE')
        member = getattr(interaction, 'member', None)
        allowed = False
        try:
            if getattr(member.guild_permissions, 'manage_guild', False):
                allowed = True
            if not allowed and verifier_role and member:
                if any(r.name == verifier_role for r in getattr(member, 'roles', [])):
                    allowed = True
        except Exception:
            allowed = False
        if not allowed:
            await interaction.response.send_message("Vous n'êtes pas autorisé à utiliser cette commande.", ephemeral=True)
            return

        # create button: since discord.py components API varies, we'll attempt a simple send with text
        try:
            await interaction.response.defer(ephemeral=True)
            if interaction.channel:
                await interaction.channel.send('Cliquez sur le bouton ci-dessous pour recevoir le message de vérification en DM. (Bouton non supporté dans cette version simplifiée)')
            else:
                await interaction.followup.send("Impossible d'envoyer le message de vérification ici (canal introuvable).", ephemeral=True)
        except Exception:
            await interaction.followup.send("Erreur lors de l'envoi du message de vérification.", ephemeral=True)
        try:
            await interaction.delete_original_response()
        except Exception:
            pass

    except Exception as err:
        logger.error(['Erreur /msgverif:', err])
        try:
            await interaction.followup.send('Erreur lors de l\'envoi du message de vérification.', ephemeral=True)
        except Exception:
            pass
